chalk-solve/src/coherence/solve.rs

```diff
@@ -85,9 +85,12 @@ impl<I: Interner> CoherenceSolver<'_, I> {
 
         let interner = self.db.interner();
 
+        let (lhs_binders, lhs_bound) = lhs.binders.as_ref().into();
+        let (rhs_binders, rhs_bound) = rhs.binders.as_ref().into();
```

Comment:
This is slightly misleading, because `as_ref` is cloning the binders, whereas it wasn't before.

Reply:
binders were cloned later anyway, but as_ref is kind of a bad name (I just copied it from rustc), should I change it?

Reply:
Oh, you mean the names, hmm, ok

Reply:
I think `as_ref` is fine. Just be sure the doc comments are clear. (The binders will almost certainly end up being an interned type instead of a vec anyways, so the clone should be cheap).